e2e/davinci-suites/src/form-fields.test.ts

```diff
@@ -1,32 +1,48 @@
-  await page.getByRole('heading', { name: 'Select Test Form' }).click();
+  await expect(page.getByRole('heading', { name: 'Select Test Form' })).toBeVisible();
   await page.getByRole('button', { name: 'Form Fields' }).click();
   await page.getByRole('textbox', { name: 'Text Input Label' }).click();
-  await page.getByRole('textbox', { name: 'Text Input Label' }).click();
-
-  const txtInput = page.getByRole('textbox', { name: 'Text Input Label' });
-  await txtInput.fill('This is some text');
-  expect(txtInput).toHaveValue('This is some text');
-
-  const flowLink = page.getByRole('button', { name: 'Flow Link' });
-  await flowLink.click();
+  await page.getByRole('textbox', { name: 'Text Input Label' }).fill('The input');
+  await expect(page.getByText('Dropdown List Label')).toBeVisible();
+  await page.locator('#dropdown-field-key').selectOption('dropdown-option1-value');
+  await page.locator('#dropdown-field-key').selectOption('dropdown-option2-value');
+  await expect(page.locator('#dropdown-field-key')).toHaveValue('dropdown-option2-value');
+  await page.getByRole('checkbox', { name: 'option1 label' }).check();
 
-  const flowButton = page.getByRole('button', { name: 'Flow Button' });
-  await flowButton.click();
+  await page.getByRole('checkbox', { name: 'option2 label' }).check();
+  await page.getByRole('checkbox', { name: 'option3 label' }).check();
+  await page.getByRole('checkbox', { name: 'option2 label' }).check();
+  await page.getByRole('button', { name: 'Select options ▼' }).click();
+  await page.locator('[id="combobox-field-key-option-option1\\ value"]').check();
+  await page.locator('[id="combobox-field-key-option-option2\\ value"]').check();
+  await page.locator('[id="combobox-field-key-option-option3\\ value"]').check();
+  await page.locator('[id="combobox-field-key-option-option1\\ value"]').uncheck();
+  await page.locator('[id="combobox-field-key-option-option3\\ value"]').uncheck();
+  await page.locator('[id="combobox-field-key-option-option1\\ value"]').check();
+  await page.getByRole('button', { name: 'Apply' }).click();
+  await expect(page.getByRole('button', { name: 'Flow Button' })).toBeVisible();
+  await expect(page.getByRole('button', { name: 'Flow Link' })).toBeVisible();
+  const requestPromise = page.waitForRequest(
+    (request) => request.url().includes('customForm') && request.method() === 'POST',
+  );
 
-  const requestPromise = page.waitForRequest((request) => request.url().includes('/customForm'));
   await page.getByRole('button', { name: 'Submit' }).click();
   const request = await requestPromise;
+
   const parsedData = JSON.parse(request.postData());
   const data = parsedData.parameters.data;
   expect(data.actionKey).toBe('submit');
   expect(data.formData).toEqual({
-    ['text-input-key']: 'This is some text',
-    ['dropdown-field-key']: '',
-    ['radio-group-key']: '',
+    'combobox-field-key': ['option1 value', 'option2 value'],
+    'checkbox-field-key': [],
```

Comment:
i'm not rendering this because the data we have ends up with the same exact text / label for this and radio

and then we have to getByAll and index through it I think or `filter` them. we can do it I just chose not to.